### DISCUSSION TITLE
Issue launching wezterm in wsl

### DISCUSSION THREAD

**Discussion by sak96** [wezterm/wezterm#4059] [Q&A] [answered]
This works:
```bash
MESA_D3D12_DEFAULT_ADAPTER_NAME=NVIDIA wezterm
```

This doesn't work
```bash
MESA_D3D12_DEFAULT_ADAPTER_NAME=NVIDIA123 wezterm
```

I wanted to use the config to set this environment variable

```lua
config.set_enviroment_variables = {
MESA_D3D12_DEFAULT_ADAPTER_NAME="NVIDIA"
}
```
this does not work. is there a better way to handle this within the config ?

**Comment by wez** [comment]
Consider using `front_end = "WebGpu"` instead of OpenGL, and you can then use https://wezfurlong.org/wezterm/config/lua/config/webgpu_preferred_adapter.html to select a preferred adapter

**Reply by sak96** [reply]
I did check out the documentation, this part. 
i tried it again.


seems like that environment variable fixes things.
without this only thing that works is using front_end="Software".

where did i get that variable from ?
[here](https://github.com/microsoft/wslg/wiki/GPU-selection-in-WSLg)

**Reply by wez** [reply]
Oh, I see. There isn't a way to tell wezterm to set that environment variable; you need to set it before you start wezterm in order for it to take effect.  You might consider setting up a shell alias or creating a custom .desktop file to launch wezterm with that set.

**Reply by sak96** [reply, marked as answer]
ok i think i will set this variable on xrdp session.

/etc/xrdp/sesman.ini
```ini
[SessionVariables]
MESA_D3D12_DEFAULT_ADAPTER_NAME="NVIDIA"
```

Thanks for the help @wez.
